common/dynamicconfig/constants.go

```diff
@@ -928,4 +928,12 @@ const (
 	WorkerStickyCacheSize = "worker.stickyCacheSize"
 	// SchedulerNamespaceStartWorkflowRPS is the per-namespace limit for starting workflows by schedules
 	SchedulerNamespaceStartWorkflowRPS = "worker.schedulerNamespaceStartWorkflowRPS"
+	// WorkerDeleteNamespaceMaxConcurrentActivityExecutionSize is the maximum concurrent activity executions the delete namespace activity worker can have
```

Comment:
Could you use a map instead of separate dynamic configs? E.g. https://github.com/temporalio/temporal/blob/main/service/worker/pernamespaceworker.go#L284

It'd be nice to factor out that code actually so anywhere we create a worker can use it

Reply:
True, though I think a comment on the dynamic config saying "this is only an activity worker, so only the activity worker settings are useful here" is enough. We might want these settings on all the workers, right? So I'd rather avoid the combinatorial explosion of workers * options

Reply:
Yeah, I was slightly concerned about the number of options as well, but we have so many already so I figured there wouldn't be much harm.

Still, for future extensibility you're right that a map is probably better, so lmk what you think of the latest version.

Reply:
I do like the map, but why can't we factor out the code from pernsworker into service/worker/common and use the same struct and conversion function in both (and future) places? Though I'm fine with putting it off to a future PR as long as the keys don't change